### PR TITLE
[NFC] Store the interface shape type in an opened element environment

### PR DESCRIPTION
**Pull request by rjmccall**

### CODE REVIEW

**Comment by rjmccall** [comment]
@swift-ci Please smoke test

**Comment by rjmccall** [comment]
@swift-ci Please test

File: lib/SIL/IR/SILPrinter.cpp [hunk 1]
```diff
@@ -2312,7 +2312,18 @@ class SILPrinter : public SILInstructionVisitor<SILPrinter> {
           << " of " << subs.getGenericSignature()
           << " at ";
     printSubstitutions(subs);
-    *this << ", shape $" << env->getOpenedElementShapeClass()
+    // The shape class in the opened environment is a canonical interface
+    // type, which won't resolve in the generic signature we just printed.
+    // Map it back to the sugared generic parameter.
+    auto sugaredShapeClass = [&] {
```
Comment (slavapestov) [inline lib/SIL/IR/SILPrinter.cpp:2318]: This is just GenericSignature::getSugaredType()

**Comment by rjmccall** [comment]
@swift-ci Please smoke test

**Pull request merged**